tooling/metamodel-generator/src/main/java/org/hibernate/processor/validation/ProcessorSessionFactory.java

```diff
@@ -631,7 +649,18 @@ private static boolean isEnumProperty(Element member) {
 
 	@Override
 	boolean isEnum(String className) {
-		final TypeElement typeElement = elementUtil.getTypeElement(className);
+		TypeElement typeElement = elementUtil.getTypeElement( className );
+		int startIdx = 0;
+		int dollarIdx;
+		while ( typeElement == null && ( dollarIdx = className.indexOf( '$', startIdx ) ) != -1 ) {
+			final String potentialBaseTypeName = className.substring( 0, dollarIdx );
+			final TypeElement potentialBaseType = elementUtil.getTypeElement( potentialBaseTypeName );
+			if ( potentialBaseType != null ) {
+				className = potentialBaseTypeName + className.substring( dollarIdx + 1 );
+				typeElement = elementUtil.getTypeElement( className );
+			}
+			startIdx = dollarIdx + 1;
+		}
```

Comment:
@beikov what is the purpose of this added code?

If I delete it, the tests still pass, and literal enum validation appears to work perfectly well in IntelliJ.

Reply:
It handles the difference between a JVM type name which uses `$` as separator for inner classes vs. Java canonical type name which uses `.` as separator. If you do e.g. `elementUtil.getTypeElement( "mypackage.MyClass$MyEnum" )`, you will get null, whereas `elementUtil.getTypeElement( "mypackage.MyClass.MyEnum" )` will return you the proper element. On the other hand, `Class.forName( "mypackage.MyClass.MyEnum" )` produces an exception, whereas `Class.forName( "mypackage.MyClass$MyEnum" )` will load the class.

Reply:
But you added a test for that, which passes even without this code.

Reply:
It works because `allowedEnumLiteralsToEnumTypeNames` will resolve to Java canonical names, which is the source for the argument passed to this method, but the method could be called from other places I guess? It's just for overall correctness.

Reply:
I mean currently this is the only place it's called. But in principle, sure, in future `isEnum()` might be called from elsewhere. But except in this case, I don't see why it would ever be called with a name that the processor API doesn't understand natively.